### DISCUSSION TITLE
[@types/leaflet] Missing/incorrect Layer options and hierarchy

### DISCUSSION THREAD

**Discussion by mikeu** [DefinitelyTyped/DefinitelyTyped#64691] [Issues with a @types package]
I'm working on converting [vue-leaflet](https://github.com/vue-leaflet/vue-leaflet) to TypeScript, and am writing a generic function with a type variable `<T extends L.Layer>`. In the function I need to work with the arbitrary layer's `options`. While this property [exists on the base class in Leaflet](https://github.com/Leaflet/Leaflet/blob/8c38dbafb614a887a546e64aff913dc7feda9a0d/src/layer/Layer.js#L30-L43),
```js
// leaflet/src/layer/Layer.js
export const Layer = Evented.extend({

	// Classes extending `L.Layer` will inherit the following options:
	options: {
		// @option pane: String = 'overlayPane'
		// By default the layer will be added to the map's [overlay pane](#map-overlaypane). Overriding this option will cause the layer to be placed on another pane by default.
		pane: 'overlayPane',

		// @option attribution: String = null
		// String to be shown in the attribution control, e.g. "© OpenStreetMap contributors". It describes the layer data and is often a legal obligation towards copyright holders and tile providers.
		attribution: null,

		bubblingMouseEvents: true
	},

        // ...
```
it is not available on the [Layer type here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/leaflet/index.d.ts#L1191-L1229).

While looking into these options because I need the `attribution` one in particular, I noticed that the `bubblingMouseEvents` property on `Layer.options` is also [not included in the `LayerOptions` interface](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/leaflet/index.d.ts#L1181-L1184), and that the options for some layer types don't extend `LayerOptions`, so are also missing these properties (for example, [`GridLayerOptions`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/leaflet/index.d.ts#L1231-L1256)).

[a comment by an automated account is not shown]

**Reply by henrythasler** [reply]
You can just create a PR (make sure to include some tests) for the typings with the changes you need.

**Comment by LeleDallas** [comment]
Same issue in React with  `@types/leaflet v1.9.12`
